sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -337,12 +337,9 @@ namespace detail {
 namespace pi {
 
 // Report error and no return (keeps compiler from printing warnings).
-// TODO: Probably change that to throw a catchable exception,
-//       but for now it is useful to see every failure.
-//
 [[noreturn]] void die(const char *Message) {
   std::cerr << "pi_die: " << Message << std::endl;
```

Comment:
I would like to hear what others think about pushing messages to std::cerr here instead of attaching it to the exception.

Reply:
Yes, me too. I followed the same way of `check_error`, i.e.,

https://github.com/intel/llvm/blob/1a8bb5331a95ad27064666c6fc4eb42f4a18e780/sycl/plugins/cuda/pi_cuda.cpp#L140-L157

Reply:
I realized I took the wrong approach modifying the `pi::die` function. Basically a function that `std::terminate` could be needed, also the other PIs have it.

In order to solve https://github.com/intel/llvm/issues/2119#issuecomment-1201548912, I made ``cuda_piextKernelSetArgMemObj`` returning an error and setting a message.
In such a way the RT will throw an exception with the specified message allowing SYCL-CTS to continue the test execution, i.e.,
```
../util/sycl_exceptions.h:252: warning:
  SYCL exception
  with category name: 'sycl'
  with code: 'runtime'
  with code raw value: '1'
  with code message: 'SYCL Error'
  with what: 'Native API failed. Native API returns: -996 (Function exists but
  address is not available)
  PI CUDA kernels only support images with channel types int32, uint32, float,
  and half.
   -996 (Function exists but address is not available)'

../util/test_base.cpp:28: FAILED:
explicitly with message:
  a SYCL exception was caught: Native API failed. Native API returns: -996
  (Function exists but address is not available)
  PI CUDA kernels only support images with channel types int32, uint32, float,
  and half.
   -996 (Function exists but address is not available)
```